nuget.cake

```diff
@@ -23,45 +22,36 @@ public static void DownloadPackages(this ICakeContext context, DirectoryPath ext
     using (var httpClient = new HttpClient())
     {
         Parallel.ForEach(
-            packageIds,
-            packageId => retryPolicy.Execute(
-                    () => context.DownloadPackage(extensionDir, httpClient, packageId)
+            packageIdAndVersionLookup,
+            packageIdAndVersion => retryPolicy.Execute(
+                () => context.DownloadPackage(extensionDir, httpClient, packageIdAndVersion)
             )
         );
     }
 }
 
-public static void DownloadPackage(this ICakeContext context, DirectoryPath extensionDir, HttpClient httpClient, string packageId)
+public static void DownloadPackage(this ICakeContext context, DirectoryPath extensionDir, HttpClient httpClient, KeyValuePair<string, string> packageIdAndVersion)
 {
-    context.Information("[{0}] fetching meta data...", packageId);
-    var packageJson = httpClient.GetStringAsync($"https://api.nuget.org/v3/registration5-semver1/{packageId.ToLowerInvariant()}/index.json")
-                    .GetAwaiter()
-                    .GetResult();
-
-    var package = LitJson.JsonMapper.ToObject<Package>(packageJson);
-
-    var packageInfo = (
-        from packageItem in package?.items ?? Enumerable.Empty<PackageItem>()
-        from item in packageItem.items ?? Enumerable.Empty<PackageRevisions>()
-        where item?.catalogEntry?.listed ?? false
-        let version = item?.catalogEntry?.version
-        let nugetVersion = version is null ? null : NuGetVersion.Parse(version)
-        orderby !(nugetVersion?.IsPrerelease ?? false), nugetVersion
-        select new
-        {
-            id = packageId,
-            version = nugetVersion?.ToNormalizedString(),
-            isPrerelease = nugetVersion?.IsPrerelease,
-            item?.packageContent
-        }
-    ).LastOrDefault();
+    var packageId = packageIdAndVersion.Key;
+    var packageVersion = packageIdAndVersion.Value;
 
-    context.Verbose("[{0}] found {1}...", packageId, packageInfo);
+    if (string.IsNullOrWhiteSpace(packageVersion))
+    {
+        context.Warning($"[{{0}}] Skipping download of {packageId} - AnalyzedPackageVersion not specified in metadata", packageId);
```

Comment:
This will skip downloading [Cake.Issues.Testing](https://github.com/cake-build/website/blob/master/extensions/Cake.Issues.Testing.yml) which doesn't have a `AnalyzedPackageVersion` in the metadata, since it is not maintained by Addin Discoverer (since it is not an extension, but helpers for writing tests). Idea was to have it included in API documentation on the website. The question is if we want to get rid of it and only handle extensions, which can be updated by  Addin Discoverer, or if we should fallback to download latest version in these cases.

Reply:
`Cake.Issues.Testing` seems to be the only exception so far. Would it make sense to manually add `AnalyzedPackageVersion` to the `Cake.Issues.Testing` YAML for now, until Addin Discoverer is updated to handle these cases as well?

I don't know the history behind why it skips this one, but I had the impression the goal of Addin Discoverer is to eventually handle all kinds of extensions to Cake, be it modules, or APIs like this one.

Reply:
`Cake.Issues.Testing` is special since it is a NuGet package which helps to write tests for addins built on top of the Cake.Issues addin. It's neither an addin, nor a module or recipe script. It's also not shown in the extension list of the website, just the api part. Before we start to implement additional logic for it I think I just delete it to make stuff simpler.